src/Chapter2.hs

```diff
@@ -605,7 +620,8 @@ Implement a function that duplicates each element of the list
 
 -}
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
+duplicate [] = []
+duplicate (x:xs) = replicate 2 x ++ duplicate xs
```

Comment:
This is the correct implementation!
However, it could be sped up a bit 🙂 

2 is not a big number, so it is manageable to right `x:x` manually.
Also, with such usage, it is possible to get rid of the list concatenation, which is slow, and use `:`.

```suggestion
duplicate (x:xs) = x : x : duplicate xs
```

Reply:
Indeed, very nice !